test/box-luatest/gh_7974_force_recovery_bugs_test.lua

```diff
@@ -0,0 +1,82 @@
+local server = require('luatest.server')
+local t = require('luatest')
+
+local g = t.group()
+
+-- Checks that force recovery works with snapshot containing no user spaces
+-- and an unknown request type.
+g.test_uknown_request_type_force_recovery = function()
+    local datadir = 'test/box-luatest/gh_7974_data/unknown_request_type'
```

Comment:
What bothers me is that it's unclear what these snapshot files contain and how they were created so one wouldn't know what to do if they want to add another test case or update the snapshots for some reason. It'd be nice to have a script that would regenerate these files, like [here](https://github.com/tarantool/tarantool/tree/master/test/xlog/upgrade/2.1.3/gh-4771-upgrade-sequence), or at least a human-readable instruction. The problem is that we don't have an xlog writer Lua API so that would require writing the generator in C. I'm fine to ignore this if everyone else thinks it's okay to leave it as is. I'd ask what @Totktonada thinks, because he worked a lot on the tests in the past.

Reply:
I'm out of context here. Can we generate the snapshot and post-process it somehow? If so, I would point to my recent [fill.lua](https://github.com/tarantool/tarantool/blob/c066f63b5f40d0a3e4c2f6d1d5232924cd7671f4/test/box-luatest/gh_4691_data/fill.lua) as an example that I feel good. If it is complicated, I would vote for an instruction for a human.

Reply:
@Totktonada we need snapshots with invalid rows to test force recovery. Basically, one needs to manually patch `checkpoint_f` to generate an invalid one. @locker suggests writing a snapshot (i.e., xrow) generation utility instead).

Reply:
It would be nice to have such a tool. While we there is no one I would write an instruction for a human.

Reply:
@Totktonada thanks for your insight!

Reply:
@locker I added snapshot generation instructions to each of the test cases, please review.